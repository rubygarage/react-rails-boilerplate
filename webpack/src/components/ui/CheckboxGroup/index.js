import React, { Component } from 'react'
import { Field } from 'redux-form'
import classNames from 'classnames'

import styles from './checkbox.css'

class CheckboxGroup extends Component {
  field = ({ input, className, meta: { touched, error }, options, align, web }) => {
    const { name, onChange, onBlur, onFocus, value: inputValue } = input

    const checkboxes = options.map(({ value, label }, index) => {
      const handleChange = (event) => {
        const arr = [...inputValue]
        if (event.target.checked) {
          arr.push(value)
        } else {
          arr.splice(arr.indexOf(value), 1)
        }
        onBlur(arr)

        return onChange(arr)
      }
      const isChecked = inputValue.includes(value)

      return (
        <label key={index} className={classNames(styles.checkbox, className, styles[web], styles[align])}>
          <input
            className={styles['checkbox-input']}
            type="checkbox"
            name={name}
            value={value}
            checked={isChecked}
            onChange={handleChange}
            onFocus={onFocus}
          />
          <span className={classNames(styles['checkbox-icon'], 'icon')} />
          <span className={styles['checkbox-text']}>{label}</span>
        </label>
      )
    })

    return (
      <div>
        <div className="flex flex-wrap">{checkboxes}</div>
        <div className="has-error">
          { touched && error && <p className="error-message">{error}</p> }
        </div>
      </div>
    )
  }

  render() {
    return <Field {...this.props} type="checkbox" component={this.field} />
  }
}

export default CheckboxGroup
