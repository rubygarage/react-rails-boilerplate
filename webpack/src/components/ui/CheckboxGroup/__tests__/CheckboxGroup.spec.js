import React from 'react'
import { shallow } from 'enzyme'
import CheckboxGroup from 'components/ui/CheckboxGroup'

describe('<CheckboxGroup />', () => {
  const defaultProps = {
    options: [{ value: 1, lable: 'Power' }, { value: 2, lable: 'Dark' }],
    className: 'type-checkbox',
    web: 'web',
    align: 'vertical',
    name: 'secondary_careers'
  }

  it('renders with empty props', () => {
    const component = shallow(<CheckboxGroup />)

    expect(component).toMatchSnapshot()
  })

  it('renders with props', () => {
    const component = shallow(<CheckboxGroup {...defaultProps} />)

    expect(component).toMatchSnapshot()
  })
})
