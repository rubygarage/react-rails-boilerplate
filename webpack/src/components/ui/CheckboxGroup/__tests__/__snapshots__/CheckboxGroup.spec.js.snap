// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxGroup /> renders with empty props 1`] = `
<Field
  component={[Function]}
  type="checkbox"
/>
`;

exports[`<CheckboxGroup /> renders with props 1`] = `
<Field
  align="vertical"
  className="type-checkbox"
  component={[Function]}
  name="secondary_careers"
  options={
    Array [
      Object {
        "lable": "Power",
        "value": 1,
      },
      Object {
        "lable": "Dark",
        "value": 2,
      },
    ]
  }
  type="checkbox"
  web="web"
/>
`;
